crates/snapshot/src/snapshotter.rs

```diff
@@ -109,10 +129,38 @@ impl<DB: Database> Snapshotter<DB> {
         }
     }
 
-    fn update_highest_snapshots_tracker(&self) {
+    /// Looks into the snapshot directory to find the highest snapshotted block of each segment, and
+    /// notifies every tracker.
+    fn update_highest_snapshots_tracker(&mut self) -> RethResult<()> {
+        // It walks over the directory and parses the snapshot filenames extracing `SnapshotSegment`
+        // and their inclusive range. It then takes the maximum block number for each specific
+        // segment.
+        for (segment, range) in std::fs::read_dir(&self.snapshots_path)
+            .map_err(|err| RethError::Custom(err.to_string()))?
+            .filter_map(Result::ok)
+            .filter_map(|entry| {
+                if let Ok(true) = entry.metadata().map(|metadata| metadata.is_file()) {
+                    return SnapshotSegment::parse_filename(&entry.file_name().to_string_lossy())
+                }
+                None
+            })
+        {
+            let max_segment_block = match segment {
+                SnapshotSegment::Headers => &mut self.highest_snapshots.headers,
+                SnapshotSegment::Transactions => &mut self.highest_snapshots.transactions,
+                SnapshotSegment::Receipts => &mut self.highest_snapshots.receipts,
+            };
```

Comment:
we can add a helper method similar to `HighestSnapshots::highest` that returns a mutable ref, so we have all these match exprs in one file

Reply:
done on #5373